### PR TITLE
Fix transform source bug 

### LINKED ISSUE #1913: Dependency update PR broke Storybook source snippets
# 🐛 Bug Report

It looks like PR #1905 was responsible for breaking Storybook source snippets. I.e. 

Screenshot 2024-03-07 at 3 57 53 PM

## 💻 Repro or Code Sample

Source snippets worked with the "Select placeholder (#1842)" PR – [Chromatic Storybook](https://60e89457a987cf003efc0a5b-xtrrplxpre.chromatic.com/)

Source snippets fail with `[object Promise]` with the "Dependency Updates March 2024 (#1905)" PR – [Chromatic Storybook](https://60e89457a987cf003efc0a5b-oqkmtpqxjr.chromatic.com/)

## 🤔 Expected Behavior

Screenshot 2024-03-07 at 4 02 59 PM

## 😯 Current Behavior

Screenshot 2024-03-07 at 3 57 53 PM

## 🌍 Your Environment

* OS & Device: macOS 
* Browser: all

### PR DESCRIPTION
**Pull request by fredvisser**
# Pull Request

## 🤨 Rationale

Fixes #1913 by using `js-beautify` instead of `prettier: ^3` in `transformSource.js`.

## 👩‍💻 Implementation

1. Leverage a popular code formatting tool - https://www.npmjs.com/package/js-beautify
2. Used the default formatting settings, as they looked correct to me.

## 🧪 Testing

Manual review of Storybook docs pages. 

## ✅ Checklist

- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

**Comment by fredvisser** [comment]
@rajsite - FYI - here is a PR with the `js-beautify` quick fix. If you have an alternative approach, feel free to kill this one.

File: packages/nimble-components/package.json [hunk 1]
```diff
@@ -139,6 +139,7 @@
     "eslint-plugin-storybook": "^0.8.0",
     "html-webpack-plugin": "^5.3.1",
     "jasmine-core": "^5.1.2",
+    "js-beautify": "^1.15.1",
     "karma": "^6.3.0",
```
Comment (rajsite) [inline packages/nimble-components/package.json:143]: Can you remove the explicit "prettier": "^3.2.5" dependency below? (we can just rely on whatever prettier-eslint brings in)
Make sure to re-run npm install after for the updated lockfile

**Review by rajsite — APPROVED** [review]

File: change/@ni-nimble-components-fd0f6918-b425-44fe-b644-17e155870de3.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-fd0f6918-b425-44fe-b644-17e155870de3.json:1]: @fredvisser: 
FYI - here is a PR with the `js-beautify` quick fix. If you have an alternative approach, feel free to kill this one.

Comment (rajsite) [reply change/@ni-nimble-components-fd0f6918-b425-44fe-b644-17e155870de3.json:1]: Yea lets ship it if it works. If we find issues can work on re-integrating prettier.

File: packages/nimble-components/.storybook/transformSource.js [hunk 3]
```diff
@@ -1,5 +1,4 @@
-import prettier from 'prettier/standalone';
-import parserHTML from 'prettier/plugins/html';
+const beautifyHTML = require('js-beautify').html;
```
Comment (rajsite) [inline packages/nimble-components/.storybook/transformSource.js:1]: Would be nice to have esm support: https://github.com/beautifier/js-beautify/issues/2199

But not a source dependency so not as big a deal breaker. Let's go!

File: packages/nimble-components/.storybook/transformSource.js [hunk 4]
```diff
@@ -78,11 +79,8 @@ export const transformSource = source => {
 
     const trimmedHTML = removeBlankLines(html);
     const emptyAttributesRemovedHTML = removeEmptyAttributes(trimmedHTML);
-    const formmattedHTML = prettier.format(emptyAttributesRemovedHTML, {
-        parser: 'html',
-        plugins: [parserHTML],
-        htmlWhitespaceSensitivity: 'ignore',
-        tabWidth: 4
+    const formmattedHTML = beautifyHTML(emptyAttributesRemovedHTML, {
+        wrap_attributes: 'force-expand-multiline'
```
Comment (rajsite) [inline packages/nimble-components/.storybook/transformSource.js:83]: FYI @fredvisser updated to this setting so attributes align better to prettier (and pprevents as much horizontal scrolling):

See without this setting (notice need horizontal scroll):
image

With this setting (in their playground, waiting for a build):
image

Previous prettier behavior:
image

**Pull request merged**